robot_smach_states/src/robot_smach_states/navigation/follow_operator2_0.py

```diff
@@ -0,0 +1,475 @@
+
+                for i in range(start, end):
+                    x = previous_point.x() + i * dx_norm * res
+                    y = previous_point.y() + i * dy_norm * res
+                    kdl_plan.append(kdl_conversions.kdlFrameStampedFromXYZRPY(x=x, y=y, z=0, yaw=yaw))
+
+            previous_point = copy.deepcopy(crumb._pose.p)
+
+        # Delete the elements from the plan within the operator radius from the robot
+        cutoff = int(self._operator_radius / (2.0 * res))
+        if len(kdl_plan) > cutoff:
+            del kdl_plan[-cutoff:]
+
+        ros_plan = frame_stampeds_to_pose_stampeds(kdl_plan)
+        # Check if plan is valid. If not, remove invalid points from the path
+        if len(ros_plan) > 0:
+            if not self._robot.base.global_planner.checkPlan(ros_plan):
+                print "Breadcrumb plan is blocked, removing blocked points"
+                # Go through plan from operator to robot and pick the first unoccupied point as goal point
+                ros_plan = [point for point in ros_plan if self._robot.base.global_planner.checkPlan([point])]
+
+        buffer_msg = Marker()
+        buffer_msg.type = Marker.POINTS
+        buffer_msg.scale.x = 0.05
+        buffer_msg.scale.y = 0.05
+        buffer_msg.header.stamp = rospy.get_rostime()
+        buffer_msg.header.frame_id = "/map"
+        buffer_msg.color.a = 1
+        buffer_msg.color.r = 1
+        buffer_msg.color.g = 0
+        buffer_msg.color.b = 0
+        # buffer_msg.lifetime = rospy.Time(.0)
+        buffer_msg.id = 0
+        buffer_msg.action = Marker.ADD
+
+        for crumb in buffer:
+            buffer_msg.points.append(kdl_conversions.kdlVectorToPointMsg(crumb.pose.frame.p))
+
+        line_strip = Marker()
+        line_strip.type = Marker.LINE_STRIP
+        line_strip.scale.x = 0.05
+        line_strip.header.frame_id = "/map"
+        line_strip.header.stamp = rospy.Time.now()
+        line_strip.color.a = 1
+        line_strip.color.r = 0
+        line_strip.color.g = 1
+        line_strip.color.b = 1
+        line_strip.id = 0
+        line_strip.action = Marker.ADD
+
+        # Push back all pnts
+        for pose_stamped in ros_plan:
+            line_strip.points.append(pose_stamped.pose.position)
+
+        self._plan_marker_pub.publish(line_strip)
+        self._breadcrumb_pub.publish(buffer_msg)
+        self._robot.base.local_planner.setPlan(ros_plan, p, o)
+        userdata.buffer_follow_out = buffer
+        return 'follow_bread'
+
+
+class AskFinalize(smach.State):
+    def __init__(self, robot):
+        smach.State.__init__(self, outcomes=['follow', 'Done'])
+        self._robot = robot
+
+    def execute(self, userdata=None):
+        sentence = "Are we there yet?"
+        self._robot.speech.speak(sentence, block=True)
+        try:
+            answer = self._robot.hmi.query(sentence, "T -> yes | no", "T")
+        except TimeoutException as e:
+            self._robot.speech.speak("I did not hear you!")
```

Comment:
If you catch this exception, 'None' is returned and the application will crash.

Furthermore, I'd get rid of the else: just de-dent the rest